Sources/StreamChat/StateLayer/ChannelList.swift

```diff
@@ -22,19 +20,24 @@ public class ChannelList {
     ) {
         self.channelListUpdater = channelListUpdater
         self.query = query
-        state = environment.stateBuilder(
-            initialChannels,
-            query,
-            dynamicFilter,
-            client.config,
-            channelListUpdater,
-            client.databaseContainer,
-            client.eventNotificationCenter
-        )
+        stateBuilder = StateBuilder {
+            environment.stateBuilder(
+                initialChannels,
+                query,
+                dynamicFilter,
+                client.config,
+                channelListUpdater,
+                client.databaseContainer,
+                client.eventNotificationCenter
+            )
+        }
     }
     
+    /// The query specifying and filtering the list of channels.
+    public let query: ChannelListQuery
+    
     /// An observable object representing the current state of the channel list.
-    public let state: ChannelListState
+    @MainActor public lazy var state: ChannelListState = stateBuilder.build()
```

Comment:
can we avoid it being lazy?

Reply:
I had an [alternative idea](https://getstream.slack.com/archives/CRB9P6L03/p1712903579989979?thread_ts=1712901708.377809&cid=CRB9P6L03) as well but it did not work for all the cases. So at the moment, I don't have ideas which would avoid using lazy var for state. `@MainActor` and lazy var are currently just tools for pushing the state creation to the main actor. Tricky